pkg/collector/corechecks/containers/docker/docker.go

```diff
@@ -90,11 +91,11 @@ func (d *DockerCheck) countAndWeightImages(sender aggregator.Sender, imageTagsBy
 		return nil
 	}
 
-	availableImages, err := du.Images(false)
+	availableImages, err := du.Images(context.TODO(), false)
```

Comment:
should we create a local context with a timeout here?

Reply:
This is in fact not needed because “per-call” timeouts are already handled inside the `docker_util` functions:

https://github.com/DataDog/datadog-agent/blob/lenaic/no_liveness_depending_on_network/pkg/util/docker/docker_util.go#L104-L105
```go
// Images returns a slice of all images.
func (d *DockerUtil) Images(ctx context.Context, includeIntermediate bool) ([]types.ImageSummary, error) {
	ctx, cancel := context.WithTimeout(ctx, d.queryTimeout)
```

In fact, for all components (kubelet, docker daemon, etc.) there is already “per call” timeouts.
The issue is that if the network is black-holing all the traffic, every single call (to docker for ex.) will block until its timeout expires and a single component loop run (of `ad-dockerlistener` for ex.) can try to make several calls.
That’s why this PR is more about adding a “global” deadline than “individual” call timeouts.